app/code/Magento/GraphQl/Controller/HttpResponse/Cors/CorsAllowMethodsHeaderProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+
+namespace Magento\GraphQl\Controller\HttpResponse\Cors;
+
+
+use Magento\Framework\App\Response\HeaderProvider\HeaderProviderInterface;
+use Magento\GraphQl\Model\Cors\ConfigurationInterface;
+
+class CorsAllowMethodsHeaderProvider implements HeaderProviderInterface
+{
+    protected $headerName = 'Access-Control-Allow-Methods';
+
+    protected $headerValue = 'GET,POST,OPTIONS';
+
```

Comment:
if this value doesn't change it's best to use static private.
But, I would even use an array, and when used in method do implode(,)
also merge it with some values in the constructor so we can make it extensible